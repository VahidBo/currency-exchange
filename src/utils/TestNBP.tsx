import React, { PropsWithChildren } from 'react';
import { NativeBaseProvider } from 'native-base';

const inset = {
  frame: { x: 0, y: 0, width: 0, height: 0 },
  insets: { top: 0, left: 0, right: 0, bottom: 0 },
};

export const TestNBP = ({ children }: PropsWithChildren<Record<string, never>>) => (
  <NativeBaseProvider initialWindowMetrics={inset}>{children}</NativeBaseProvider>
);
