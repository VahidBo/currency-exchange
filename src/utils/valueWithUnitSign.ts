import { Currency } from 'src/components';

export function valueWithUnitSign(value: number, currency: Currency): string {
  return { USD: `$${value}`, EUR: `€${value}`, GBP: `£${value}` }[currency];
}
