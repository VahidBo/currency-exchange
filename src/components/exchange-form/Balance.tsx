import React from 'react';
import { Currency } from './types';
import { Badge, Text } from 'native-base';
import { valueWithUnitSign } from '../../utils';

export interface BalanceProps {
  value: number;
  currency: Currency;
}

export function Balance({ value, currency }: BalanceProps) {
  return (
    <Badge
      colorScheme="info"
      rounded="full"
      mb={2}
      borderColor="indigo.700"
      width="80%"
      height={12}
      justifyContent="center"
    >
      <Text fontSize="xl" color="indigo.900">
        Balance:{' '}
        <Text bold fontSize="xl">
          {valueWithUnitSign(value, currency)}
        </Text>
      </Text>
    </Badge>
  );
}
