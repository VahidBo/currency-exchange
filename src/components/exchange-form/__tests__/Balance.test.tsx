import React from 'react';
import renderer from 'react-test-renderer';
import { Balance } from '../Balance';
import { TestNBP } from '../../../utils/TestNBP';

test('Snapshot of Balance with 200 USD', () => {
  const tree = renderer
    .create(
      <TestNBP>
        <Balance value={200} currency="USD" />
      </TestNBP>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Snapshot of Balance with 150 EUR', () => {
  const tree = renderer
    .create(
      <TestNBP>
        <Balance value={150} currency="EUR" />
      </TestNBP>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
