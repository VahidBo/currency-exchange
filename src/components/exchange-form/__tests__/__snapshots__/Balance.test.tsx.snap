// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of Balance with 150 EUR 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    colorScheme="info"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#bae6fd",
        "borderColor": "#4338ca",
        "borderRadius": 9999,
        "borderWidth": 1,
        "height": 48,
        "justifyContent": "center",
        "marginBottom": 8,
        "paddingBottom": 2,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 2,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Object {
          "backgroundColor": undefined,
          "color": "#312e81",
          "fontFamily": undefined,
          "fontSize": 20,
          "fontStyle": "normal",
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 30,
          "textDecorationLine": undefined,
        }
      }
    >
      Balance:
       
      <Text
        style={
          Object {
            "backgroundColor": undefined,
            "fontFamily": undefined,
            "fontSize": 20,
            "fontStyle": undefined,
            "fontWeight": "700",
            "textDecorationLine": undefined,
          }
        }
      >
        â‚¬150
      </Text>
    </Text>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Snapshot of Balance with 200 USD 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    colorScheme="info"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#bae6fd",
        "borderColor": "#4338ca",
        "borderRadius": 9999,
        "borderWidth": 1,
        "height": 48,
        "justifyContent": "center",
        "marginBottom": 8,
        "paddingBottom": 2,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 2,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Object {
          "backgroundColor": undefined,
          "color": "#312e81",
          "fontFamily": undefined,
          "fontSize": 20,
          "fontStyle": "normal",
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 30,
          "textDecorationLine": undefined,
        }
      }
    >
      Balance:
       
      <Text
        style={
          Object {
            "backgroundColor": undefined,
            "fontFamily": undefined,
            "fontSize": 20,
            "fontStyle": undefined,
            "fontWeight": "700",
            "textDecorationLine": undefined,
          }
        }
      >
        $200
      </Text>
    </Text>
  </View>
</RNCSafeAreaProvider>
`;
